lib/Factory/MenuItemFactory.php

```diff
@@ -38,7 +38,43 @@ protected function build(WP_Post $item, Menu $menu) : CoreInterface {
 		return $class::build($item, $menu);
 	}
 
-	protected function get_menuitem_class($item, $menu) : string {
+	protected function get_menuitem_class(WP_Post $item, Menu $menu) : string {
+		/**
+		 * Filters the class(es) used for different menu items.
+		 *
+		 * Read more about this in the documentation for [Menu Item Class Maps](https://timber.github.io/docs/v2/guides/class-maps/#the-menu-item-class-map).
+		 *
+		 * The default Menu Item Class Map will contain class names for locations that map to `Timber\MenuItem`.
+		 *
+		 * @since 2.0.0
+		 * @example
+		 * ```
+		 * add_filter( 'timber/menuitem/classmap', function( $classmap ) {
+		 *     $custom_classmap = [
+		 *         'primary'   => MenuItemFooter::class,
+		 *         'secondary' => MenuItemHeader::class,
+		 *     ];
+		 *
+		 *     return array_merge( $classmap, $custom_classmap );
+		 * } );
+		 * ```
+		 *
+		 * @param array $classmap The menu item class(es) to use. An associative array where the key is
+		 *                        the location and the value the name of the class to use for this
+		 *                        menu item or a callback that determines the class to use.
+		 */
+		$classmap = apply_filters( 'timber/menuitem/classmap', [] );
```

Comment:
While working with 2.x on a project, I found that it could be quite useful to have nav location based classmaps for menu items.